Update module_utils/asa.py with NetworkModule

Qalthos opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

module_utils/asa.py
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
2.2
```
##### SUMMARY

Going along with EOS, IOS, et al., this module follows the same general pattern, so has been updated accordingly.

Qalthos committed:
Rewrite asa.py for NetworkModule

Qalthos committed:
Common CliBase run_commands implementation

privateip commented:
looks good.  one minor update, remove abort_config and commit_config methods.  Those were removed from netcfg.Config and are implemented on a device by device basis.  Should be good to merge after that

Qalthos commented:
I was also thinking of moving all the `raise NotImplementedError` into CliBase so that it wouldn't have to be specified if missing (at least for Cli-based implementations)

Qalthos committed:
Remove abort_config and commit_config

Also move unimplemented command methods to CliBase, where they can be overridden
as needed
